docker/docker_client.go

```diff
@@ -258,6 +271,41 @@ func newDockerClient(ctx *types.SystemContext, ref dockerReference, write bool,
 	}, nil
 }
 
+// CheckAuth validates credentials
+func CheckAuth(ctx context.Context, sCtx *types.SystemContext, username, password, registry string) error {
```

Comment:
(The function name or the comment could be a tiny bit more descriptive, e.g. saying that it validates them by trying to log in into the registry (as opposed to only verifying that they are non-empty or the like).)

Reply:
fixed